.github/workflows/upgrade-python-requirements.yml

```diff
@@ -0,0 +1,68 @@
+name: Upgrade Requirements
+
+on:
+  schedule:
+    # will start the job at 00:15 UTC every monday
+    - cron: "15 0 * * 1"
+  workflow_dispatch:
+    inputs:
+      branch:
+        description: "Target branch to create requirements PR against"
+        required: true
+        default: 'master'
+
+jobs:
+  upgrade_requirements:
+    runs-on: ubuntu-20.04
+
+    strategy:
+      matrix:
+        python-version: ["3.8"]
+
+    steps:
+      - name: setup target branch
+        run: echo "target_branch=$(if ['${{ github.event.inputs.branch }}' = '']; then echo 'master'; else echo '${{ github.event.inputs.branch }}'; fi)" >> $GITHUB_ENV
+
+      - uses: actions/checkout@v1
+        with:
+          ref: ${{ env.target_branch }}
+      
+      - name: setup python
+        uses: actions/setup-python@v2
```

Comment:
Question, is `actions/setup-python@v2` going to mean we use python 2.7, or using python 3.8?

Reply:
this means we'll be using `setup-python` action's version 2 not python 2, as you can see below we set up python version
`python-version: ${{ matrix.python-version }}`